bpf/lib/nodeport.h

```diff
@@ -1988,7 +1993,7 @@ static __always_inline int rev_nodeport_lb4(struct __ctx_buff *ctx, int *ifindex
 		bpf_mark_snat_done(ctx);
 
 		*ifindex = ct_state.ifindex;
-#ifdef TUNNEL_MODE
+#if defined(TUNNEL_MODE) && __ctx_is != __ctx_xdp
```

Comment:
This is just being extra defensive but not strictly required, right? As the agent will forbid tunneling + LB acceleration (or are there other cases where XDP can be enabled)?

Reply:
It's needed so that we can skip compiling the encap-section when inside XDP, no?

Reply:
The XDP and the tunneling mode are mutually exclusive (the agent has a check for it). I added this to avoid referencing the `encap_redirect` label which is undefined if `ENABLE_EGRESS_GATEWAY` is undefined, but `TUNNEL_MODE` is defined. Not sure how realistic is this datapath configuration though, but `cd bpf/ && make` caught it.